client/my-sites/plugins/plugin-list-header/index.jsx

```diff
@@ -126,52 +111,54 @@ export default React.createClass( {
 			const actionBarVisible = actionBarDomElement.offsetHeight <= MAX_ACTIONBAR_HEIGHT;
 			this.setState( { actionBarVisible } );
 		}, 1 );
-	},
+	}
 
 	showPluginTooltip() {
 		this.setState( { addPluginTooltip: true } );
-	},
+	}
 
 	hidePluginTooltip() {
 		this.setState( { addPluginTooltip: false } );
-	},
+	}
 
 	toggleBulkManagement() {
 		this.props.toggleBulkManagement();
 
 		this.maybeMakeActionBarVisible();
-	},
+	}
 
 	onBrowserLinkClick() {
 		analytics.ga.recordEvent( 'Plugins', 'Clicked Add New Plugins' );
-	},
+	}
 
 	canUpdatePlugins() {
 		return this.props.selected.some( plugin => plugin.sites.some( site => site.canUpdateFiles ) );
-	},
+	}
 
 	unselectOrSelectAll() {
-		const someSelected = this.props.selected.length > 0;
-		this.props.setSelectionState( this.props.plugins, ! someSelected );
+		const { plugins, selected } = this.props;
+		const someSelected = selected.length > 0;
+		this.props.setSelectionState( plugins, ! someSelected );
 		analytics.ga.recordEvent( 'Plugins', someSelected ? 'Clicked to Uncheck All Plugins' : 'Clicked to Check All Plugins' );
-	},
+	}
 
 	renderCurrentActionButtons() {
-		const isJetpackSelected = this.props.selected.some( plugin => 'jetpack' === plugin.slug ),
-			needsRemoveButton = this.props.selected.length && this.canUpdatePlugins() && ! isJetpackSelected,
-			buttons = [],
-			rightSideButtons = [],
-			leftSideButtons = [],
-			autoupdateButtons = [],
-			activateButtons = [];
+		const { translate, selected } = this.props;
+		const isJetpackSelected = selected.some( plugin => 'jetpack' === plugin.slug );
+		const needsRemoveButton = selected.length && this.canUpdatePlugins() && ! isJetpackSelected;
```

Comment:
Some of these statements are logic that is duplicated in several locations. Would it make sense to create methods for such cases? Just like it's being done for `this.canUpdatePlugins()`.